libcaf_net/test/udp_datagram_socket.cpp

```diff
@@ -30,85 +30,68 @@
+    addresses = local_addresses("localhost");
+    CAF_CHECK(!addresses.empty());
+    ep = ip_endpoint(*addresses.begin(), 0);
     auto send_pair = unbox(make_udp_datagram_socket(ep));
-    auto send_socket = send_pair.first;
+    send_socket = send_pair.first;
     auto receive_pair = unbox(make_udp_datagram_socket(ep));
-    auto receive_socket = receive_pair.first;
+    receive_socket = receive_pair.first;
     ep.port(ntohs(receive_pair.second));
-    CAF_MESSAGE("sending data to: " << to_string(ep));
-    auto send_guard = make_socket_guard(send_socket);
-    auto receive_guard = make_socket_guard(receive_socket);
-    if (auto err = nonblocking(socket_cast<net::socket>(receive_socket), true))
-      CAF_FAIL("nonblocking returned an error" << err);
-    auto test_read_res = read(receive_socket, make_span(buf));
-    if (auto err = get_if<sec>(&test_read_res))
-      CAF_CHECK_EQUAL(*err, sec::unavailable_or_would_block);
-    else
-      CAF_FAIL("read should have failed");
-
-    auto write_ret = write(send_socket, as_bytes(make_span(hello_test)), ep);
-    if (auto num_bytes = get_if<size_t>(&write_ret))
-      CAF_CHECK_EQUAL(*num_bytes, hello_test.size());
-    else
-      CAF_FAIL("write returned an error: " << sys.render(get<sec>(write_ret)));
+  }
 
-    auto read_ret = read(receive_socket, make_span(buf));
-    if (auto read_res = get_if<std::pair<size_t, ip_endpoint>>(&read_ret)) {
-      CAF_CHECK_EQUAL(read_res->first, hello_test.size());
-      buf.resize(read_res->first);
-    } else {
-      CAF_FAIL("write returned an error: " << sys.render(get<sec>(read_ret)));
-    }
-    string_view buf_view{reinterpret_cast<const char*>(buf.data()), buf.size()};
-    CAF_CHECK_EQUAL(buf_view, hello_test);
+  ~fixture() {
+    close(send_socket);
+    close(receive_socket);
   }
 
+  std::vector<ip_address> addresses;
   actor_system_config cfg;
   actor_system sys{cfg};
-  ip_address v4_local;
-  ip_address v6_local;
-  std::vector<ip_address> addrs;
+  ip_endpoint ep;
+  udp_datagram_socket send_socket;
+  udp_datagram_socket receive_socket;
 };
 
 } // namespace
 
 CAF_TEST_FIXTURE_SCOPE(udp_datagram_socket_test, fixture)
 
 CAF_TEST(send and receive) {
-  // TODO: check which versions exist and test existing versions accordingly
-  // -> local_addresses()
-  if (contains(v4_local)) {
-    test_send_receive(v4_local);
-  } else if (contains(v6_local)) {
-    test_send_receive(v6_local);
-  } else {
-    CAF_FAIL("could not resolve 'localhost'");
-  }
+  std::vector<byte> buf(1024);
+  if (auto err = nonblocking(socket_cast<net::socket>(receive_socket), true))
+    CAF_FAIL("nonblocking returned an error" << err);
+  CAF_CHECK_EQUAL(read(receive_socket, make_span(buf)),
+                  sec::unavailable_or_would_block);
+  CAF_MESSAGE("sending data to " << to_string(ep));
+  CAF_CHECK_EQUAL(write(send_socket, as_bytes(make_span(hello_test)), ep),
+                  hello_test.size());
+  int rounds = 0;
```

Comment:
It's unclear from the context what a "round" is supposed to be here IMO. I think renaming the variable makes the intent clearer:

```suggestion
  int receive_attempts = 0;
```